### PR TITLE
Registration agent

### PR DESCRIPTION
**Pull request by isalyne34**
- Add poc registration to scheduler

@Razano26 @hugoponthieu

### CODE REVIEW

**Comment by hugoponthieu** [comment]
Thanks for bringing the code here. The goal will be to clean this very experimental code in order to be ready to handle better every situation and be more customizable.

**Comment by hugoponthieu** [comment]
+ client that connect to scheduler
+ store in a global var the id of the agent
+ log for starting server

File: agent/src/registering_service.rs [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+use crate::AGENT_ID;
+
+mod proto {
+    tonic::include_proto!("scheduler");
+}
+
+pub async fn register_agent(url: &String) -> Result<(), Box<dyn std::error::Error>> {
+    match proto::agent_client::AgentClient::connect(url.to_string()).await {
+        Ok(mut cli) => {
+            //TODO: Use real health data
```
Comment (theotchlx) [inline agent/src/registering_service.rs:12]: Is this `//TODO` done?

File: agent/src/registering_service.rs [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+            let req = proto::Health {
+                cpu_usage: 1,
+                memory_usage: 1,
+            };
+            let request = tonic::Request::new(req);
+            let response: RegisterAgentResponse = cli.register_agent(request).await?.into_inner();
+            let mut agent_id = AGENT_ID.lock()?;
+            *agent_id = response.id.parse::<i32>()?;
```
Comment (theotchlx) [inline agent/src/registering_service.rs:20]: agent_id should be an u32, not an i32

File: api/proto/agent/registration.proto
Comment (theotchlx) [inline api/proto/agent/registration.proto:1]: this file is outdated, maybe the branch needs to be rebased on its parent branch?

File: agent/src/registering_service.rs (hunk 1 above)
Comment (hugoponthieu) [reply agent/src/registering_service.rs:12]: It will be address in later PRs

File: api/proto/agent/registration.proto
Comment (hugoponthieu) [reply api/proto/agent/registration.proto:1]: Done !

File: agent/src/registering_service.rs (hunk 2 above)
Comment (hugoponthieu) [reply agent/src/registering_service.rs]: Done !

**Review by Razano26 — APPROVED** [review]
Lgtm 🚀

**Pull request merged**